### PR TITLE
UI - show table/view detail #1996

### PR DESCRIPTION
**Pull request by maheshsapkal**

### CODE REVIEW

**Review by rymurr — COMMENTED** [review]
Thanks @maheshsapkal this looks good.

Could you please:
1) post screenshots summarizing the changes
2) fix the todos in tests
3) don't use linter skip comments (or add a comment to justify why they are being used)

**Comment by maheshsapkal** [comment]
@rymurr please find screenshots below for content view, for different types.
DeltaLakeContents
IcebergContent
SqlViewContent

**Comment by maheshsapkal** [comment]
> Thanks @maheshsapkal this looks good.
> 
> Could you please:
> 
> 1. post screenshots summarizing the changes
> 2. fix the todos in tests
> 3. don't use linter skip comments (or add a comment to justify why they are being used)

@rymurr 
1. Please find the above screenshot.
2. Fixed todo in tests
3. Fixed typescript error and removed linter skip comments.

File: ui/src/Explore/Components/ContentsView.test.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  }, 5000);
+  expect(asFragment()).toMatchSnapshot();
+  expect(screen.getByText("VIEW")).toBeInTheDocument();
+});
+
+const iceBergContent: Content = {
+  type: "ICEBERG_TABLE",
+  id: "5d8bdfec-96d7-44f6-8770-1ffcf09e044d",
+  metadataLocation: "/path/to/metadata/",
+  idGenerators: "6c3563f6847feda9",
```
Comment (ajantha-bhat) [inline ui/src/Explore/Components/ContentsView.test.tsx:95]: Also as per new release `idGenerators` is removed. 

please use the latest ICEBERG_TABLE content.

Comment (ajantha-bhat) [reply ui/src/Explore/Components/ContentsView.test.tsx:95]: https://github.com/projectnessie/nessie/pull/2626

Comment (maheshsapkal) [reply ui/src/Explore/Components/ContentsView.test.tsx:95]: @ajantha-bhat Looks like the following fields are newly added, do we have to show them on UI? it seems like those are internal purposes only but wanted to make sure.
 schemaId, specId, sortOrderId
 cc: @nastra

Comment (nastra) [reply ui/src/Explore/Components/ContentsView.test.tsx:95]: let's omit them for now. We can always add them later to the UI if necessary

Comment (maheshsapkal) [reply ui/src/Explore/Components/ContentsView.test.tsx]: @ajantha-bhat idGenerators is removed from test cases. Also updated latest content for same.

File: ui/src/Explore/Components/ContentView.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        </div>
+      </div>
+    );
+  };
+
+  const renderInvalidView = () => {
+    return <EmptyMessageView message="Invalid type" />;
+  };
+
+  const renderContents = () => {
```
Comment (nastra) [inline ui/src/Explore/Components/ContentView.tsx:137]: renderContents -> renderContent

File: ui/src/Explore/Components/ContentView.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    switch (tableContents.type) {
```
Comment (nastra) [inline ui/src/Explore/Components/ContentView.tsx:138]: tableContents -> tableContent

File: ui/src/Explore/Components/ContentView.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+              <span className={"ml-2"}>{metadataLocation}</span>
+            </div>
+          )}
+        </div>
+        {imageView("/iceberg.png")}
+      </div>
+    );
+  };
+
+  const renderView = (sqlViewContents: SqlView) => {
```
Comment (nastra) [inline ui/src/Explore/Components/ContentView.tsx:107]: sqlViewContents -> sqlViewContent

File: ui/src/Explore/Components/ContentView.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+              <span className={"ml-2"}>{lastCheckpoint}</span>
+            </div>
+          )}
+        </div>
+        {imageView("/delta.png")}
+      </div>
+    );
+  };
+
+  const renderIcebergView = (icebergContents: IcebergTable) => {
```
Comment (nastra) [inline ui/src/Explore/Components/ContentView.tsx:86]: icebergContents -> icebergContent

File: ui/src/Explore/Components/ContentView.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+  const imageView = (imgPath: string) => {
+    return (
+      <div className={"ContentView__imagetWrapper"}>
+        <img alt="IcebergTable" src={`${imgPath}`} width="180" height="180" />
+      </div>
+    );
+  };
+
+  const renderDeltaView = (deltaContents: DeltaLakeTable) => {
```
Comment (nastra) [inline ui/src/Explore/Components/ContentView.tsx:33]: deltaContents -> deltaContent

File: ui/src/Explore/Components/ContentView.test.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+};
+
+const sqlViewContent: Content = {
+  type: "VIEW",
+  id: "ae63654e-e692-49b2-8cd6-21e0d980893c",
+  sqlText: "CREATE VIEW test.v AS SELECT * FROM t",
+  dialect: Dialect.Hive,
+};
+
+const contentsList = {
```
Comment (nastra) [inline ui/src/Explore/Components/ContentView.test.tsx:121]: contentsList -> contentList

File: ui/src/Explore/TableListing.tsx [hunk 8]
```diff
@@ -67,51 +80,120 @@ const entryToKey = (entry: Entry): Key => {
   return {
     name: entry.name.elements[entry.name.elements.length - 1],
     type: entry.type === "UNKNOWN" ? "CONTAINER" : "TABLE",
+    contentsKey: entry.name,
```
Comment (nastra) [inline ui/src/Explore/TableListing.tsx:83]: contentsKey -> contentKey

File: ui/src/Explore/TableListing.tsx [hunk 9]
```diff
@@ -67,51 +80,120 @@ const entryToKey = (entry: Entry): Key => {
...
+    .then((data) => {
+      return data;
+    })
+    .catch((e) => log.error("getMultipleContents", e));
+};
+
 interface Key {
   name: string;
   type: "CONTAINER" | "TABLE";
+  contentsKey: ContentKey;
```
Comment (nastra) [inline ui/src/Explore/TableListing.tsx:107]: contentsKey -> contentKey

**Review by nastra — COMMENTED** [review]
still a few renames that should be done

File: ui/src/Explore/Components/ContentsView.tsx [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+  const renderIcebergView = (icebergContents: IcebergTable) => {
+    const { id, metadataLocation, idGenerators } = icebergContents;
+    return (
+      <div className={"ContentsView"}>
+        <div className={"ContentsView__contentWrapper"}>
+          <div>
+            <span>Type: </span>
+            <span className={"ml-2"}> {tableContents.type}</span>
+          </div>
+          {id && (
```
[the remark this answers is not in the document]
Comment (maheshsapkal) [reply ui/src/Explore/Components/ContentsView.tsx]: > still a few renames that should be done

Done

**Review by nastra — APPROVED** [review]
LGTM

**Pull request merged**